homeassistant/components/braviatv/__init__.py

```diff
@@ -110,178 +115,171 @@ def __init__(
-            for cmd in command:
-                self.braviarc.send_command(cmd)
-
-    def _get_source(self) -> str | None:
-        """Return the name of the source."""
-        for key, value in self.content_mapping.items():
-            if value == self.content_uri:
-                return key
-        return None
-
-    def _refresh_volume(self) -> bool:
-        """Refresh volume information."""
-        volume_info = self.braviarc.get_volume_info(self.audio_output)
-        if volume_info is not None:
-            volume = volume_info.get("volume")
-            self.volume_level = volume / 100 if volume is not None else None
-            self.audio_output = volume_info.get("target")
-            self.min_volume = volume_info.get("minVolume")
-            self.max_volume = volume_info.get("maxVolume")
-            self.muted = volume_info.get("mute", False)
-            return True
-        return False
-
-    def _refresh_channels(self) -> bool:
-        """Refresh source and channels list."""
-        if not self.source_list:
-            self.content_mapping = self.braviarc.load_source_list()
-            self.source_list = []
-            if not self.content_mapping:
-                return False
-            for key in self.content_mapping:
-                if key not in self.ignored_sources:
-                    self.source_list.append(key)
-        return True
-
-    def _refresh_playing_info(self) -> None:
-        """Refresh playing information."""
-        playing_info = self.braviarc.get_playing_info()
-        program_name = playing_info.get("programTitle")
-        self.channel_name = playing_info.get("title")
-        self.program_media_type = playing_info.get("programMediaType")
-        self.content_uri = playing_info.get("uri")
-        self.source = self._get_source()
-        self.duration = playing_info.get("durationSec")
-        if not playing_info:
-            self.channel_name = "App"
-        if self.channel_name is not None:
-            self.media_title = self.channel_name
-            if program_name is not None:
-                self.media_title = f"{self.media_title}: {program_name}"
-        else:
-            self.media_title = None
+    def _sources_extend(self, sources: dict, source_type: str) -> None:
+        """Extend source map and source list."""
+        for item in sources:
+            item["type"] = source_type
+            title = item.get("title")
+            uri = item.get("uri")
+            if not title or not uri:
+                continue
+            self.source_map[uri] = item
+            if title not in self.ignored_sources:
+                self.source_list.append(title)
 
-    def _update_tv_data(self) -> None:
-        """Connect and update TV info."""
-        power_status = self.braviarc.get_power_status()
+    async def _async_update_data(self) -> None:
+        """Connect and fetch data."""
+        power_status = await self.client.get_power_status()
```

Comment:
All client request calls should raise an exception if the request fails. So we should handle possible exceptions in all these client calls.

To not have to repeat the code that handles the exception in all service handlers etc we often use a decorator.

Example:
https://github.com/home-assistant/core/blob/b7b965c9c9575e3717eb4bef39809c55c700b7d4/homeassistant/components/vlc_telnet/media_player.py#L50-L68

https://github.com/home-assistant/core/blob/b7b965c9c9575e3717eb4bef39809c55c700b7d4/homeassistant/components/vlc_telnet/media_player.py#L242-L245